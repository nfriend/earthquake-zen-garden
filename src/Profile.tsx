import React, { Fragment } from 'react';
import PageTitle from './PageTitle';

const data = require('./data.json');

function Nav(): JSX.Element {
  const avatarAltText = `${data.profile.firstName} ${data.profile.lastName}'s profile image`;

  const profileDetails: { label: string; value: string }[] = [
    { label: 'First name', value: data.profile.firstName },
    { label: 'Last name', value: data.profile.lastName },
    { label: 'Phone', value: data.profile.phone },
    { label: 'Email', value: data.profile.email },
    { label: 'Bio', value: data.profile.bio },
  ];

  return (
    <div className="flex flex-col">
      <PageTitle title="Profile" />
      <div className="flex flex-col sm:flex-row sm:justify-center items-center sm:items-start">
        <img
          src={data.profile.avatarImage}
          alt={avatarAltText}
          className="h-56 mb-4 sm:mb-0 object-scale-down"
        />
        <div
          className="grid grid-cols-3 max-w-sm mx-4"
          data-testid="profile-details"
        >
          {profileDetails.map(({ label, value }) => {
            return (
              <Fragment key={label}>
                <h3 className="font-bold mb-2 mr-2">{label}</h3>
                <span className="col-span-2">{value}</span>
              </Fragment>
            );
          })}
        </div>
      </div>
    </div>
  );
}

export default Nav;
