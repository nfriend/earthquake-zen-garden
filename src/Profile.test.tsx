import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Profile from './Profile';

const data = require('./data.json');

describe('Profile.tsx', () => {
  beforeEach(() => {
    render(<Profile />);
  });

  test('renders the avatar image', () => {
    const avatarElement = screen.getByAltText(
      `${data.profile.firstName} ${data.profile.lastName}'s profile image`,
    );
    expect(avatarElement).toBeInTheDocument();
  });

  test('renders profile details', () => {
    const profileDetails = screen.getByTestId('profile-details');
    expect(profileDetails).toMatchSnapshot();
  });
});
