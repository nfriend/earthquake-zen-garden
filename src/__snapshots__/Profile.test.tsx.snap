// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.tsx renders profile details 1`] = `
<div
  class="grid grid-cols-3 max-w-sm mx-4"
  data-testid="profile-details"
>
  <h3
    class="font-bold mb-2 mr-2"
  >
    First name
  </h3>
  <span
    class="col-span-2"
  >
    Sally
  </span>
  <h3
    class="font-bold mb-2 mr-2"
  >
    Last name
  </h3>
  <span
    class="col-span-2"
  >
    Wang
  </span>
  <h3
    class="font-bold mb-2 mr-2"
  >
    Phone
  </h3>
  <span
    class="col-span-2"
  >
    01-343-989-2345
  </span>
  <h3
    class="font-bold mb-2 mr-2"
  >
    Email
  </h3>
  <span
    class="col-span-2"
  >
    sw@nowhere.ic.kp
  </span>
  <h3
    class="font-bold mb-2 mr-2"
  >
    Bio
  </h3>
  <span
    class="col-span-2"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur
  </span>
</div>
`;
